lib/traject/macros/transformation.rb

```diff
@@ -157,6 +157,26 @@ def gsub(pattern, replace)
           acc.collect! { |v| v.gsub(pattern, replace) }
         end
       end
+
+      def delete_if(arg)
+        p = if arg.respond_to? :include?
+              proc { |v| arg.include?(v) }
+            else
+              proc { |v| arg === v }
+            end
+
+        ->(_, acc) { acc.delete_if(&p) }
+      end
+
+      def select(arg)
+        p = if arg.respond_to? :include?
```

Comment:
Drat, I thought Array#=== already did what we wanted, but you're right it doesn't! Oh well, not quite as pretty an implementation. I'm not the only one to have wondered. 

Good job keeping the somewhat expensive `respond_to?` outside the inner proc, thanks!

Reply:
Yeah, I thought about using Array.to_set but after me going on about scope creep and benchmarking, I decided not to.

But, I guess we could just force `.to_set` on any Array?  It would still not be as clean as if we assume everything can respond to `===`.